src/bin/sail.ml

```diff
@@ -526,6 +527,15 @@ let parse_config_file file =
     Reporting.warn "" Parse_ast.Unknown (Printf.sprintf "Failed to parse configuration file: %s" message);
     None
 
+exception VersionParseError of string
+
+(* Same as int_of_string but it details the string that couldn't be parsed. *)
+let helpful_int_of_string s =
+  match int_of_string_opt s with Some i -> i | None -> raise (VersionParseError ("Error parsing " ^ s))
```

Comment:
The OCaml `Option` module contains functions for combining functions that return options.